framework/src/main/java/org/checkerframework/common/value/ValueCheckerUtils.java

```diff
@@ -389,4 +401,129 @@ public static int getMinLen(Tree tree, ValueAnnotatedTypeFactory valueAnnotatedT
         AnnotatedTypeMirror minLenType = valueAnnotatedTypeFactory.getAnnotatedType(tree);
         return valueAnnotatedTypeFactory.getMinLenValue(minLenType);
     }
+
+    /**
+     * Optimize the given JavaExpression, using Value Checker annotations.
```

Comment:
`Optimize` is a vague name. I think I'm fine with it here as the name of this method - it conveys the right intuition for what this method does - but this Javadoc needs to define it. I think the specific meaning here is "constant fold any parts of the given expression that the factory has exact values for", based on reading the code.